spark/v3.2/spark/src/main/java/org/apache/iceberg/spark/source/SparkScan.java

```diff
@@ -202,21 +202,13 @@ public boolean supportColumnarReads(InputPartition partition) {
     }
   }
 
-  static long numFilesToScan(CombinedScanTask scanTask) {
-    long fileCount = 0L;
-    for (FileScanTask file : scanTask.files()) {
-      fileCount += 1L;
-    }
-    return fileCount;
-  }
-
   private static class RowReader extends RowDataReader implements PartitionReader<InternalRow> {
     private long numSplits;
 
     RowReader(ReadTask task) {
       super(task.task, task.table(), task.expectedSchema(), task.isCaseSensitive());
-      numSplits = numFilesToScan(task.task);
-      LOG.debug("reading {} file splits for table {} using RowReader", numSplits, task.table().name());
+      numSplits = (long) task.task.files().size();
```

Comment:
Do we really need to typecast this to long? Isn't `task.task.files().size();` enough?

Reply:
`java.util.Collection#size()` returns an `int`. Yes, without the explicit cast, the compiler will do it for us as `numsplits` is declared to be a `long`.
If an explicit cast is worse for performance, I shall remove it. Otherwise, it is a matter of style.

Reply:
I don't think it really makes a difference, the implicit cast should be just as good so I would probably just remove the cast

Reply:
I'll remove the cast.